code/game/objects/items/nemesis_items.dm

```diff
@@ -0,0 +1,458 @@
+/obj/item/shield/energy/nemesis/afterattack(atom/target, mob/user, proximity, click_parameters)
+	if(!istype(target, /obj/item/clothing/gloves/rapid/nemesis))
+		return ..()
+
+	var/obj/item/clothing/gloves/rapid/nemesis/gloves = target
+	if(gloves.shield != src)
+		return
+
+	if(active)
+		active = !active
+		icon_state = "[base_icon_state][active]"
+		force = initial(force)
+		w_class = WEIGHT_CLASS_TINY
+		playsound(user, 'sound/weapons/saberoff.ogg', 35, TRUE)
+		slowdown = 0
+
+	playsound(src, 'sound/mecha/mechmove03.ogg', 50, TRUE)
+	user.dropItemToGround(src, TRUE)
+	forceMove(gloves)
+
+/obj/item/nemesis_trap
+	name = "disk trap"
+	desc = "A small disk with an energy clamp, designed to be placed on the walls."
+	icon = 'icons/obj/nemesis.dmi'
+	icon_state = "trap"
+	w_class = WEIGHT_CLASS_SMALL
+	item_flags = NO_PIXEL_RANDOM_DROP | NOBLUDGEON
+
+	var/active = FALSE
+	var/obj/item/clothing/glasses/hud/security/sunglasses/nemesis/glasses
+	var/list/beams = list()
+
+/obj/item/nemesis_trap/afterattack(atom/target, mob/user, proximity, click_parameters)
+	if(!istype(target, /turf/closed))
+		return ..()
+
+	to_chat(user, "<span class='notice'>You begin setting [src] up...</span>")
+	if(!do_after(user, 3 SECONDS, src))
+		return ..()
+
+	if(ishuman(user))
+		var/mob/living/carbon/human/owner = user
+		if(istype(owner.get_item_by_slot(ITEM_SLOT_EYES), /obj/item/clothing/glasses/hud/security/sunglasses/nemesis))
+			glasses = owner.get_item_by_slot(ITEM_SLOT_EYES)
+
+	active = TRUE
+	icon_state = "trap_deployed"
+	dir = get_dir(target, user)
+	user.dropItemToGround(src, TRUE)
+	update_icon()
+	anchored = TRUE
+	playsound(src, 'sound/magic/blind.ogg', 50, TRUE)
+	deploy_beams()
+
+/obj/item/nemesis_trap/attack_hand(mob/living/user, list/modifiers)
+	if(!active)
+		return ..()
+
+	to_chat(user, "<span class='warning'>You attempt to disarm [src]...</span>")
+	var/disarm_time = 15 SECONDS
+	if(ishuman(user))
+		var/mob/living/carbon/human/disarmer = user
+		if(istype(disarmer.get_item_by_slot(ITEM_SLOT_GLOVES), /obj/item/clothing/gloves/rapid/nemesis))
+			disarm_time = 1 SECONDS
+
+	if(!do_after(user, disarm_time, src))
+		return
+
+	to_chat(user, "<span class='notice'>You successfully disarm [src].</span>")
+	qdel(src)
+
+/obj/item/nemesis_trap/attackby(obj/O, mob/user, params)
+	if(!active)
+		return ..()
+
+	if(!istype(O, /obj/item/screwdriver))
+		return ..()
+
+	to_chat(user, "<span class='warning'>You attempt to disarm [src]...</span>")
+	var/disarm_time = 3 SECONDS
+	if(ishuman(user))
+		var/mob/living/carbon/human/disarmer = user
+		if(istype(disarmer.get_item_by_slot(ITEM_SLOT_GLOVES), /obj/item/clothing/gloves/rapid/nemesis))
+			disarm_time = 1 SECONDS
+
+	if(!do_after(user, disarm_time, src))
+		return
+
+	to_chat(user, "<span class='notice'>You successfully disarm [src].</span>")
+	qdel(src)
+
+/obj/item/nemesis_trap/proc/deploy_beams()
+	for(var/obj/item/nemesis_trap/trap in view(3, get_turf(src)))
+		if(trap == src || !trap.active)
```

Comment:
what happens if you set up two pairs of these traps next to each other? does each node send a beam to each other node?

Reply:
Thanks for the reminder, fixed that